hadoop-ozone/dist/src/main/compose/upgrade/upgrades/manual-upgrade/test.sh

```diff
@@ -20,32 +20,37 @@
 
 set -e -o pipefail
 
-COMPOSE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-export COMPOSE_DIR
-
-: "${OZONE_REPLICATION_FACTOR:=3}"
-: "${OZONE_UPGRADE_FROM:="0.5.0"}"
-: "${OZONE_UPGRADE_TO:="1.0.0"}"
-: "${OZONE_VOLUME:="${COMPOSE_DIR}/data"}"
-
-export OZONE_REPLICATION_FACTOR OZONE_UPGRADE_FROM OZONE_UPGRADE_TO OZONE_VOLUME
-
-source "${COMPOSE_DIR}/testlib.sh"
-
-create_data_dir
-
-prepare_for_binary_image "${OZONE_UPGRADE_FROM}"
-load_version_specifics "${OZONE_UPGRADE_FROM}"
-first_run
-unload_version_specifics
-
-from=$(get_logical_version "${OZONE_UPGRADE_FROM}")
-to=$(get_logical_version "${OZONE_UPGRADE_TO}")
-execute_upgrade_steps "$from" "$to"
-
-prepare_for_binary_image "${OZONE_UPGRADE_TO}"
-load_version_specifics "${OZONE_UPGRADE_TO}"
-second_run
-unload_version_specifics
-
-generate_report
+# Fail if required vars are not set.
+set -u
+: "${OZONE_UPGRADE_FROM}"
+: "${OZONE_UPGRADE_TO}"
+: "${TEST_DIR}"
+: "${OZONE_UPGRADE_CALLBACK}"
+set +u
+
+source "$TEST_DIR"/compose/non-ha/load.sh
+source "$TEST_DIR"/testlib.sh
+[[ -f "$OZONE_UPGRADE_CALLBACK" ]] && source "$OZONE_UPGRADE_CALLBACK"
```

Comment:
Is the intention here to fail if the callback file does not exist?

Reply:
No, actually the opposite. The callback file should be optional since all the functions in it are optional. This way the framework can be used to quickly try out an upgrade between two versions without defining an empty file if no callbacks are desired. The `callback` function used to invoke the callbacks will print a message for each undefined callback it skips.

Reply:
But given `set -e`, I think the above would fail if the file does not exist.  To make it optional:

```bash
[[ ! -f "$OZONE_UPGRADE_CALLBACK" ]] || source "$OZONE_UPGRADE_CALLBACK"
```

(or use `if`).

Reply:
I tested it without a callback file and it worked, but I have updated the sourcing lines to your suggestion to be safe.

Reply:
No, sorry, you are right:

> The shell does not exit if the command that fails is ... part of a && or || list

Reply:
Ok thanks, I reverted the change since I think the original reads better and the functionality will be identical.